### PR TITLE
Update of V6 for fastsim and QCD smear samples

### PR DESCRIPTION
**Pull request by zhenbinwu**
These are the current code I am using for fastsim and QCD smearing samples. I think they are tested

### CODE REVIEW

File: python/modules/DeepTopProducer.py [hunk 1]
```diff
@@ -97,9 +97,10 @@ def beginFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
             tTagEffFileName = self.taggerWD + "/tTagEff_%(era)s.root"%{"era":self.era}
     
             if self.isFastSim:
-                sample = os.path.splitext(os.path.basename(inputFile.GetName()))[0]
+                sample = os.path.splitext(os.path.basename(inputFile.GetName()))[0].split("_split")[0]
             else:
-                sample = self.sampleName
+                # For QCD smear, using the orignal QCD
+                sample = self.sampleName.replace("Smear_", "")
```
Comment (pastika) [inline python/modules/DeepTopProducer.py]: I guess other samples don't contain "Smear_" ...

File: python/modules/BtagSFWeightProducer.py [hunk 2]
```diff
@@ -62,7 +62,7 @@ def beginFile(self, inputFile, outputFile, inputTree, wrappedOutputTree):
             del self.h_eff_udsg
 
         if self.FastSim:
-            sampleName = os.path.splitext(os.path.basename(inputFile.GetName()))[0]
+            sampleName = os.path.splitext(os.path.basename(inputFile.GetName()))[0].split("_split")[0]
```
Comment (pastika) [inline python/modules/BtagSFWeightProducer.py]: what is this fixing?

File: python/modules/DeepTopProducer.py (hunk 1 above)
Comment (zhenbinwu) [reply python/modules/DeepTopProducer.py]: Yeah, in the current cfg, only QCD smear sample has "Smear". If other samples don't contain "Smear", the replace() won't affect the sampleName.

File: python/modules/BtagSFWeightProducer.py (hunk 2 above)
Comment (zhenbinwu) [reply python/modules/BtagSFWeightProducer.py]: This is used for fastsim signals needed to be splited when the file is too large and run for over 2 days. The splitting script is also added as "SplitNanoAOD.py"

**Comment by zhenbinwu** [comment]
Can we merge this PR?

**Comment by pastika** [comment]
PR looks ok to me

**Comment by scarletnorberg** [comment]
Looks fine to me also.

**Pull request merged**